BotProject/Templates/CSharp/Scripts/Node/botProjectDeploy.ts

```diff
@@ -0,0 +1,506 @@
+        const settings = await fs.readJSON(settingsPath)
+        const luisSettings = settings.luis
+
+        let luisEndpointKey: string
+
+        if (!luisAuthoringKey) {
+            luisAuthoringKey = luisSettings.authoringKey
+            luisEndpointKey = luisSettings.endpointKey
+        }
+
+        if (!luisAuthoringRegion) {
+            luisAuthoringRegion = luisSettings.region
+        }
+
+        if (!language) {
+            language = 'en-us'
+        }
+
+        if (luisAuthoringKey && luisAuthoringRegion) {
+            // publishing luis
+            const remoteBotPath = path.join(publishFolder, 'ComposerDialogs')
+            const botFiles = await this.getFiles(remoteBotPath)
+            const modelFiles = botFiles.filter((name) => {
+                return name.endsWith('.lu') && this.notEmptyLuisModel(name)
+            })
+
+            const generatedFolder = path.join(remoteBotPath, 'generated')
+            if (!await fs.pathExists(generatedFolder)) {
+                await fs.mkdir(generatedFolder)
+            }
+            let builder = new luBuild.Builder(msg => console.log(msg))
+
+            const loadResult = await builder.loadContents(modelFiles,
+                language || '',
+                environment || '',
+                luisAuthoringRegion || '')
+
+            const buildResult = await builder.build(
+                loadResult.luContents,
+                loadResult.recognizers,
+                luisAuthoringKey,
+                luisAuthoringRegion,
+                name,
+                environment,
+                language,
+                false,
+                loadResult.multiRecognizers,
+                loadResult.settings)
+            await builder.writeDialogAssets(buildResult, true, generatedFolder)
+
+            console.log(`lubuild succeed`)
+
+            const luisConfigFiles = (await this.getFiles(remoteBotPath)).filter(filename => filename.includes('luis.settings'))
+            let luisAppIds: any = {}
+
+            for (let luisConfigFile of luisConfigFiles) {
+                const luisSettings = await fs.readJson(luisConfigFile)
+                Object.assign(luisAppIds, luisSettings.luis)
+            }
+
+            const luisEndpoint = `https://${luisAuthoringRegion}.api.cognitive.microsoft.com`
+            let luisConfig: any = {
+                endpoint: luisEndpoint,
+                endpointKey: luisEndpointKey
+            }
+
+            Object.assign(luisConfig, luisAppIds)
+
+            const deploymentSettingsPath = path.join(publishFolder, 'appsettings.deployment.json')
+            let settings: any = await fs.readJson(deploymentSettingsPath)
+            settings['luis'] = luisConfig
+
+            await fs.writeJson(deploymentSettingsPath, settings)
+            const token = await this.creds.getToken()
+
+            const getAccountUri = `${luisEndpoint}/luis/api/v2.0/azureaccounts`
```

Comment:
Is there a luis client that we could use for this instead of crafting urls and requests manually? So that if the luis API changes, the client does the changes for us :)

Reply:
> Is there a luis client that we could use for this instead of crafting urls and requests manually? So that if the luis API changes, the client does the changes for us :)

yes, I should use that client, and indeed there is an azure luis client inside the @azure-js-sdk, but it also have some issues, I have tried to use that client for luis accounts authoring but failed, so I turned to use the straight forward http request way..